### DISCUSSION TITLE
Generate TOC for binder from ASN?

### DISCUSSION THREAD

**Discussion by vestingz** [jonaswinkler/paperless-ng#126] [Feature Requests]
Hi,
thank you for this nice project! :)

My question is rather workflow related: is it possible to generate some sort of table of contents from ASN e.g. 1 - 343 to have some sort of backup index in case paperless goes down? I'd like to print this TOC and put it together with all the documents in the binder then.

Thank you again!

**Comment by jonaswinkler** [comment]
Very good idea!

I'd have to look into how to actually create PDF documents from raw text / formatted text. I'd also have to consider where to put that in the UI so that it does not get in the way.

However, since this project is apparently picking up some momentum, I'd like to address more pressing issues first.

**Reply by vestingz** [reply]
Cool, thanks for considering it :)
On a first glance, maybe it could just simply be a combination of [enscript](https://www.gnu.org/software/enscript/enscript.html) and `ps2pdf` coming with ghostscript: raw text --> convert to ps with enscript --> convert ps to pdf via ps2pdf.

**Reply by amenk** [reply]
> I'd have to look into how to actually create PDF documents from raw text / formatted text. I'd also have to consider where to put that in the UI so that it does not get in the way.

There is a similar request here: https://github.com/jonaswinkler/paperless-ng/discussions/1016#discussioncomment-741286

I think we do not need to create actual PDF documents - probably just some print-css needs to be applied to the document list? And the document list needs a way to filter by ASN ranges. What do you think?

**Reply by jonaswinkler** [reply]
The API already supports this:

`http://paperless:8000/api/documents/?archive_serial_number__gte=1&archive_serial_number__lte=100&ordering=archive_serial_number&page_size=100`

Maybe you can use that to write a script of some sort?

**Reply by amenk** [reply]
Interesting. But gte/lte querying is not yet possible via the UI, right?

**Reply by amenk** [reply]
Advanced search: `asn:[1 to 100]` works :-)

**Reply by jonaswinkler** [reply]
Fascinating! I'll try to sort out the sorting issue.

**Reply by amenk** [reply]
https://github.com/jonaswinkler/paperless-ng/pull/1019

**Comment by amenk** [comment]
This is should now be possible since 1.4.3

1. Use advanced search to search for an asn range by entering for example `asn:[1 to 100]`
2. Sort the list by the criteria you need
3. Print the list using the web-browsers print function

Remark / room for improvement: If the result has multiple pages, you to print each page separately.
